lms/djangoapps/course_api/blocks/tests/test_api.py

```diff
@@ -77,3 +79,21 @@ def test_sub_structure(self):
                 self.assertIn(unicode(block.location), blocks['blocks'])
             else:
                 self.assertNotIn(unicode(block.location), blocks['blocks'])
+
+    def test_filtering_by_block_types(self):
+        sequential_block = self.store.get_item(self.course.id.make_usage_key('sequential', 'sequential_y1'))
+
+        # not filtered blocks
+        blocks = get_blocks(self.request, sequential_block.location, self.user)
+        self.assertEquals(len(blocks['blocks']), 5)
+        found_not_problem = False
+        for key in blocks['blocks']:
```

Comment:
Nit: Instead of using a regex, do an equality check against `block['type']`.
So:
```
for block in blocks['blocks']:
    if block['type'] != 'problem'
```